piet-direct2d/src/d2d.rs

```diff
@@ -146,6 +155,30 @@ impl From<Error> for piet::Error {
     }
 }
 
+impl From<PathGeometry> for Geometry {
```

Comment:
I'm open to other ideas on cleaner/shorter ways to achieve this upcast.

Reply:
I think this is ok. From a performance perspective, upcasts in COM are basically free, but downcasts require QueryInterface.

Reply:
Yeah, I just meant in terms of lines of code.

Reply:
By COM standards involving any kind of subclassing, I've seen a lot worse :)